django/cantusdb_project/main_app/management/commands/add_differentiae.py

```diff
@@ -0,0 +1,89 @@
+from main_app.models import Chant, Differentia
+from django.core.management.base import BaseCommand
+
+
+# Python script to sync the differentia database with CantusDB. Previously, we had a CharField that represented
+# the differentia ID. Now, we change this field to a ForeignKey that points to the new Differentia model that contains
+# the ID, melodic transcription, and mode.
+
+
+DIFFERENTIA_DATA = "differentia_data.txt"
+
+
+class Command(BaseCommand):
+    def handle(self, *args, **options):
+        differentia_database = get_differentia_data(DIFFERENTIA_DATA)
+        total = len(differentia_database)
+        count = 0
+        for item in differentia_database:
+            differentia_id, melodic_transcription, mode, _ = item
+            differentia = Differentia.objects.filter(
+                differentia_id=differentia_id
+            ).first()
+
+            if differentia:
+                differentia.melodic_transcription = melodic_transcription
+                differentia.mode = mode
+                differentia.save()
+            else:
+                differentia = Differentia(
+                    differentia_id=differentia_id,
+                    melodic_transcription=melodic_transcription,
+                    mode=mode,
+                )
+                differentia.save()
```

Comment:
I don't really understand why we need an if/else here... are there duplicate IDs in `differentia_data.txt`? In what situations is `differentia` expected to evaluate to `True`?

Reply:
Maybe, in the `if` block, an error message would be more appropriate than allowing this unexpected situation to pass silently and just overwrite?

Reply:
This is checking if a `Differenita` object already exists with that ID. Say, if you ran the script more than once, you don't want duplicate objects in the database.